### ISSUE TITLE
Cypress run starts but does not run any tests

### ISSUE THREAD

**Issue by cushdan** [cypress-io/github-action#1154]
When we run our cypress tests they appear to start and find the specs but then no tests are run. The step ends and reports success and the cypress cloud link that is presented in the logs is an old run. Here is the output from the step:

```
[STARTED] Task without title.
2024-03-17T23:09:14.798Z @cypress/github-action found command "/usr/local/bin/yarn"
2024-03-17T23:09:14.798Z @cypress/github-action with arguments start
2024-03-17T23:09:14.798Z @cypress/github-action running "/usr/local/bin/yarn" start in /home/runner/work/projectname-ui/projectname-ui
2024-03-17T23:09:14.798Z @cypress/github-action waiting for the command to finish? false
/usr/local/bin/yarn start
yarn run v1.22.22
$ next start --hostname local.projectname.com --port 2000
ready - started server on local.projectname.com:2000, url: http://local.projectname.com:2000
info  - Loaded env from /home/runner/work/projectname-ui/projectname-ui/.env
warn  - The `target` config is deprecated and will be removed in a future version.
See more info here https://nextjs.org/docs/messages/deprecated-target-config
[SUCCESS] Task without title.

Opening Cypress...

DevTools listening on ws://127.0.0.1:34471/devtools/browser/909ce3cb-d39e-4b4e-9f63-fd0a77713073
[3742:0317/230918.590670:ERROR:object_proxy.cc(590)] Failed to call method: org.freedesktop.portal.Settings.Read: object_path= /org/freedesktop/portal/desktop: unknown error type: 

====================================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:        13.6.3                                                                         │
  │ Browser:        Electron 114 (headless)                                                        │
  │ Node Version:   v20.8.1 (/home/runner/runners/2.314.1/externals/node20/bin/node)               │
  │ Specs:          24 found (login.spec.ts, status-check.spec.ts, institution/search-org.spec.ts, │
  │                  institution/setup-organization.spec.ts, organization/add-folder.spec.ts, orga │
  │                 nization/assign-transaction.spec.ts, organization/search-subaccount.spec.ts, s │
  │                 ubaccount/close-subacco...)                                                    │
  │ Searched:       cypress/e2e/**/*spec.ts                                                        │
  │ Params:         Tag: false, Group: false, Parallel: true                                       │
  │ Run URL:        https://cloud.cypress.io/projects/<projectid>/runs/334                              │
  │ Experiments:    experimentalOriginDependencies=true                                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


====================================================================================================

  (Run Finished)
2024-03-17T23:09:29.328Z @cypress/github-action Cypress tests: 0 failed
2024-03-17T23:09:29.328Z @cypress/github-action resultsUrl https://cloud.cypress.io/projects/<projectid>/runs/334
2024-03-17T23:09:29.329Z @cypress/github-action all done, exiting
```

This is the step definition from our workflow

```yml
      - name: Cypress run
        uses: cypress-io/github-action@v6
        with:
          start: yarn start
          install: false
          record: true
          parallel: true # Runs test in parallel using settings above
        env:
          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY2 }}
          DEBUG: '@cypress/github-action'
```

**Comment by MikeMcC399** [comment]
@cushdan

Could you post your full workflow?

Have you provided a matrix of containers for the parallel run like:

```yml
    strategy:
      fail-fast: false
      matrix:
        containers: [1, 2, 3]
```

Is this a private repository or is it public?

**Comment by cushdan** [comment]
Yes we do have a matrix defined as seen in the full workflow below. This is a private repo. Thanks!

```
name: Cypress Tests

on:
  pull_request:
    branches: [ main ]
  schedule:
    # Run daily at 6pm EST 23 UTC
    - cron: '0 23 * * *'
  workflow_dispatch:

jobs:
  cypress-run:
    runs-on: ubuntu-22.04
    permissions:
      contents: read
      id-token: write
    strategy:
      fail-fast: false # https://github.com/cypress-io/github-action/issues/48
      matrix:
        containers: [1, 2, 3, 4] # Uses 4 parallel instances

    steps:
      - name: Checkout the Code
        uses: actions/checkout@v4

      - name: Assume AWS Decrypt Role
        uses: aws-actions/configure-aws-credentials@v4
       
      - name: Decrypt Secrets
       
      ## workaround to support assuming different roles within the same workflow
      ## https://github.com/aws-actions/configure-aws-credentials/issues/236#issuecomment-881033383
      - name: reset creds
        shell: bash

      - name: Assume AWS E2E Role
        uses: aws-actions/configure-aws-credentials@v4
        with:

      - name: Setup Node v20
        uses: actions/setup-node@v4
        with:
          node-version: 20

      - name: Install OpenSSL
        run: sudo apt-get update && sudo apt-get install -y openssl
      - name: Generate private key & self-signed certificate
        run: openssl req -x509 -newkey rsa:4096 -keyout localhost-key.pem -out localhost.pem -days 365 -subj "/CN=localhost/CN=local.projectname.com" -nodes

      - name: Add local.foobar.com to /etc/hosts
        run: echo "127.0.0.1 local.projectname.com" | sudo tee -a /etc/hosts

      - name: Install NGINX
        run: sudo apt-get update && sudo apt-get install -y nginx
      - name: Start NGINX
        run: sudo /usr/sbin/nginx -c ${GITHUB_WORKSPACE}/nginx/conf/nginx.conf

      - name: Build projectname
        run: |
          yarn --frozen-lockfile
          yarn build

      - name: Cypress run
        uses: cypress-io/github-action@v6
        with:
          start: yarn start
          install: false
          record: true
          parallel: true # Runs test in parallel using settings above
        env:
          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY2 }}
          DEBUG: '@cypress/github-action'
```

**Comment by cushdan** [comment]
also, worth noting that it seems to show this behavior only when the action is triggered from the schedule, but that could be a coincidence

**Comment by MikeMcC399** [comment]
@cushdan 

I don't see anything obviously wrong with the workflow. I can't comment on the permissions part though as I have never tried to lock this down.

Have you tried without parallel recording? In other words, commenting out the following:

```yml
#    strategy:
#      fail-fast: false # https://github.com/cypress-io/github-action/issues/48
#      matrix:
#        containers: [1, 2, 3, 4] # Uses 4 parallel instances
```
...

```yml
#          record: true
#          parallel: true # Runs test in parallel using settings above
```

In parallel recording Cypress Cloud should be telling each container what test it is supposed to run so that it does load-balancing. It may be failing in this part.

**Comment by cushdan** [comment]
ok, are there any additional logs that could help narrow things down? I've added DEBUG: to the cypress step already. 

It seems strange that the url it provides at the end of the run is one from 4 days ago. Could it be getting confused if there are no new commits and it's determining what to do based on the commit sha? I could try creating a custom build id from the gh action run id

**Comment by MikeMcC399** [comment]
@cushdan 

The documentation advises

> We recommend passing the GITHUB_TOKEN secret (created by the GH Action automatically) as an environment variable. This will allow correctly identifying every build and avoid confusion when re-running a build.

```
        env:
          # pass GitHub token to allow accurately detecting a build vs a re-run build
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

You might want to try that in addition to the other test I suggested.

**Comment by cushdan** [comment]
I made added the `GITHUB_TOKEN` back to the step and ran into some permissions errors. I think I may have resolved it by expanding the requested permission scopes for the token to essentially give `read-all` access as well as the `id-token:write` that we relied on.

```
permissions:
      actions: read
      checks: read
      contents: read
      deployments: read
      issues: read
      discussions: read
      packages: read
      pages: read
      pull-requests: read
      repository-projects: read
      security-events: read
      statuses: read
      id-token: write
```

I do see a difference in the build metadata that the cypress step is using when doing the run so I think the lack of permissions was limiting what the action was able to do.

It ran successfully tonight and we'll see how the next few days go!
